velox/type/DecimalUtil.h

```diff
@@ -80,31 +80,30 @@ class DecimalUtil {
     }
   }
 
-  template <typename TOutput>
-  inline static std::optional<TOutput> rescaleBigint(
+  template <typename TInput, typename TOutput>
+  inline static std::optional<TOutput> rescaleInt(
       const int128_t inputValue,
       const int toPrecision,
       const int toScale) {
     static_assert(
         std::is_same_v<TOutput, UnscaledShortDecimal> ||
         std::is_same_v<TOutput, UnscaledLongDecimal>);
+    static_assert(
```

Comment:
Any reason for this check? We should be able to handle `int16_t` also right?

Reply:
Handling all int types now. Please review.